src/channel/fs_prover_channel.rs

```diff
@@ -0,0 +1,112 @@
+use crate::channel::{Channel, ChannelStates, ProverChannel}; // ProverChannel 추가
+use crate::hashutil::TempHashContainer;
+use ark_ff::Field;
+use std::marker::PhantomData;
+
+use rand_chacha::rand_core::RngCore;
+use rand_chacha::ChaCha20Rng;
+
+pub struct FSProverChannel<F: Field, H: TempHashContainer> {
+    _ph: PhantomData<(F, H)>,
+    prng: ChaCha20Rng,
+    proof: Vec<u8>,
+    states: ChannelStates,
+}
+
+impl<F: Field, H: TempHashContainer> AsMut<ChannelStates> for FSProverChannel<F, H> {
+    fn as_mut(&mut self) -> &mut ChannelStates {
+        &mut self.states
+    }
+}
+
+impl<F: Field, H: TempHashContainer> AsRef<ChannelStates> for FSProverChannel<F, H> {
+    fn as_ref(&self) -> &ChannelStates {
+        &self.states
+    }
+}
+
+
+impl<F: Field, H: TempHashContainer> Channel for FSProverChannel<F, H> {
+    type Field = F;
+
+    fn recv_felem(&mut self, felem: Self::Field) -> Result<Self::Field, anyhow::Error> {
+        // TODO
+        Ok(felem)
+    }
+
+    fn recv_bytes(&mut self, n: usize) -> Result<Vec<u8>, anyhow::Error> {
+        assert!(
+            !self.is_query_phase(),
+            "Prover can't receive randomness after query phase has begun."
+        );
+
+        let mut bytes = vec![0u8; n];
+        self.prng.fill_bytes(&mut bytes);
+        Ok(bytes)
+    }
+
+    // receive random number from verifier
+    fn random_number(&mut self, upper_bound: u64) -> u64 {
+        assert!(
+            !self.is_query_phase(),
+            "Prover can't receive randomness after query phase has begun."
+        );
+
+        // TODO : change bytes size dynamically
+        let mut raw_bytes = [0u8; std::mem::size_of::<u64>()];
+        self.prng.fill_bytes(&mut raw_bytes);
+        let number = u64::from_le_bytes(raw_bytes);
+
+        assert!(
+            upper_bound < 0x0001_0000_0000_0000,
+            "Random number with too high an upper bound"
+        );
+
+        number % upper_bound
+    }
+
+    fn random_field(&mut self) -> Self::Field {
+        assert!(
+            !self.is_query_phase(),
+            "Prover can't receive randomness after query phase has begun."
+        );
+
+        let mut raw_bytes = [0u8; std::mem::size_of::<u64>()];
```

Comment:
This is not enough bytes to sample a random field element :)

We also need to have exactly the same behavior as the stone-prover channel(s).

So we need to implement this:

https://github.com/starkware-libs/stone-prover/blob/main/src/starkware/channel/noninteractive_prover_channel.cc#L74